### ISSUE TITLE
zk list --created date, returns notes from one day after if YAML date does not have timestamp

### ISSUE THREAD

**Issue by tjex** [zk-org/zk#382]
### Check if applicable

- [X] I have searched the existing issues (**required**)
- [X] I'm willing to help fix the problem and contribute a pull request

### Describe the bug

I created a daily note today (17th Jan). The system metadata via Finder shows that the file was created today, and the template for naming my daily notes uses `{{format-date now}}`, yet when I run `zk list --created=2024-01-17` the note is not returned.

It is returned if I run `zk list --created=2024-17-16`

```bash
$ zk list --created=2024-01-16
2024-01-17 d/2024/2024-01-17.md (16 hours ago)
```
Proof of date creation via `ll -l`

```bash
.rw-r--r-- tjex staff 935B Wed 17 Jan 2024-01-17.md
```

### How to reproduce?

1. Setup daily note conf as instructed [here](https://github.com/zk-org/zk/blob/main/docs/daily-journal.md)
2. create daily note and save
3. run `zk list --created=<the date today in YYYY-MM-DD format>
4. see if the daily note is returned or not

### zk configuration

```toml
[group.daily]
paths = ["d/2024"]

[group.daily.note]
filename = "{{format-date now}}"
extension = "md"

[alias]
daily = 'zk new --no-input "$ZK_NOTEBOOK_DIR/d/2024"'
```

### Environment

```bash
zk 0.14.0
system: Darwin 22.6.0 arm64
```

**Comment by tjex** [comment]
Well... Its strange. If I replicate the steps in my testing vault, it works as expected. Will need to investigate (much) further.

**Comment by tjex** [comment]
Getting this on `zk new` 

`zk: warning: sql: Scan error on column index 2, name "source_path": converting NULL to string is unsupported`

But again, not in the test vault, so this will be something in my config I'm guessing

**Comment by tjex** [comment]
Deleted and regenerated my notebook.md. This fixed the error message. But whenever I create a new note, and print out the data on it, it states it was created 17 hours ago (the same as in the original case of my daily note)

**Comment by tjex** [comment]
If I copy the file from my own vault, to the testing vault, and run `zk index`, the file that was created today *also* reports that it was created (now) 17 hours ago.

This is despite the 'bug' not being present in the testing vault whereby I can create a note with `zk new` and it will show up in `zk list --created=2024-01-17`.

If I run the same process in my own vault, the note will not be returned under notes created on 2024-01-17.

This is super weird...

**Comment by tjex** [comment]
So, I've narrowed it down to my custom default template.

For some reason this template: 
```markdown
---
title: {{title}}
date: {{format-date now}}
id: {{id}}
tags: [seed]
aliases:
---

# {{title}}

{{content}}


---
References:

```

Is creating a 'created date' as seen by zk at a fixed time (as the time elapsed is always increasing) , currently notes being created are given the elapsed time of '18 hours ago'. 

The standard default template does not produce this error.
I can create a note and it will report being created 'just now' when queried.

**Comment by tjex** [comment]
It was the `{{format-date now}}` being included in the frontmatter.

A note created with this template, will return the correct creation date. 

```markdown
---
title: {{title}}
id: {{id}}
tags: [seed]
aliases:
---

# {{title}}

{{content}}


---
References:
```

Will look into this further

**Comment by mickael-menu** [comment]
That's right, the note creation date is taken from, in order of precedence:

1. `date` in the YAML frontmatter
2. file creation date on the file system
3. the current time as a fallback

I thought it made sense to use the `date` from the YAML frontmatter, but this is arguable.

Maybe there's an issue in the parsing or output of the `{{format-date now}}`?

**Comment by tjex** [comment]
Had my latest and final update half written but a guest arrived for dinner.

Long story short, I was expecting something that was of course not to be expected...

Of course `2024-01-17` will be seen as the moment that we ticked over from the 16th to the 17th of January, which will be `x` amount of hours ago 🤦‍♂️

Holy hell... How annoying.

The sql error was also because I had old sql queries left over in some of my notes from my time in Obsidian with dataview.
So there was some weird stuff going on when sql indexed the files I'm guessing. So double face palm 🤦‍♂️🤦‍♂️

**Comment by tjex** [comment]
... but maybe it actually isn't a bug, but a design flaw in the time library?
Or a design flaw in the time elapsed library we use (forgot the name)

Date today = 2024-01-18

```bash
zk list --created=2024-01-16
2024-01-17 d/2024/2024-01-17.md (yesterday)
```

So `--created=2024-01-16` looks like it means "all notes from 2024-01-16 00:00 till 2024-01-17 00:00". But this is confusing as I would expect to be shown notes from 2024-01-16 00:00 till 2024-01-17 23:59` when asking for ntoes created on 2024-01-16

**Comment by tjex** [comment]
It would appear like the date in YAML isn't being respected at all.

I manually created a note now at 22:37, and gave it the date as if it were made 1 minute past midnight today. Then ran `zk index`

```text
sys > ./zk list --created=today
2024-01-18 00:00:00 +0100 CET <- fmt.Println(start of day value)
delete delete.md (1 minute ago)
```

```markdown
---
date: 2024-01-18 00:01
---

# delete
```

If I create a note via `zk new`, it clearly takes the YAML date, but that also shows that something weird is going on as despite having created the note, I get the elapsed time of 'not yet)

> I've run the cmd again 10 minutes later at 22:45, and it's still returning 'not yet'

```text
sys > ./zk list --created=today
2024-01-18 00:00:00 +0100 CET
delete 0mr2.md (not yet)

  ‣ # delete
```

```markdown
---
title: delete
date: 2024-01-18 22:35
id: 0mr2
tags: [seed]
aliases: []
---

# delete
```

**Comment by tjex** [comment]
closed with #384
